src/test/java/core/basesyntax/service/impl/OperationStrategyImplTest.java

```diff
@@ -0,0 +1,76 @@
+package core.basesyntax.service.impl;
+
+import static core.basesyntax.db.Storage.storage;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import core.basesyntax.dao.FruitDao;
+import core.basesyntax.dao.impl.FruitDaoImpl;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class OperationStrategyImplTest {
+    private static FruitDao fruitDao;
+    private OperationStrategy operationStrategy;
+    private Map<FruitTransaction.Operation, OperationStrategy> operationHandlers = new HashMap<>();
+
+    @BeforeAll
+    static void beforeAll() {
+        fruitDao = new FruitDaoImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        operationHandlers.put(FruitTransaction.Operation.BALANCE, new BalanceOperation(fruitDao));
+        operationHandlers.put(FruitTransaction.Operation.SUPPLY, new SupplyOperation(fruitDao));
+        operationHandlers.put(FruitTransaction.Operation.PURCHASE, new PurchaseOperation(fruitDao));
+        operationHandlers.put(FruitTransaction.Operation.RETURN, new ReturnOperation(fruitDao));
+        operationStrategy = new OperationStrategyImpl(operationHandlers);
+        storage.clear();
+    }
+
+    @Test
+    void balanceOperation_addsNewFruitWithQuantity() {
```

Comment:
```suggestion
    void balanceOperation_addsNewFruitWithQuantity_ok() {
```
use the same pattern